src/primitives/uint.ts

```diff
@@ -12,10 +27,28 @@ export class UnsignedInteger {
     return false;
   }
 
+  static toUint32(num: number): Uint32 {
+    if (this.isUint32(num)) {
+      return num;
+    }
+    throw new TypeError();
+  }
+
   static toUint64(num: bigint): Uint64 {
     if (this.isUint64(num)) {
       return num;
     }
     throw new TypeError();
   }
+
+  static upCast(num: Uint32): Uint64 {
+    return BigInt(num) as Uint64;
+  }
+
+  static downCast(num: Uint64): Uint32 {
```

Comment:
do we need this?

Reply:
Not right now no, can remove if you prefer

Reply:
Oh but if you're sure we'll need it why not